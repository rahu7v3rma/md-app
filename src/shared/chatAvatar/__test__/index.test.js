import React from 'react';
import renderer, { act } from 'react-test-renderer';

import ChatAvatar from '@/shared/chatAvatar';

let tree;
beforeEach(async () => {
    tree = await act(() => renderer.create(<ChatAvatar />));
});

describe('test chatAvatar', () => {
    it('snapshot test', async () => {
        expect(tree).toMatchSnapshot();
    });

    it('renders with default values', async () => {
        expect(tree.root).toBeTruthy();
    });

    it('renders with a provided image path', async () => {
        await act(() => tree.update(<ChatAvatar path="1234" />));
        expect(tree.root.findByProps({ imageId: '1234' })).toBeTruthy();
    });

    it('displays online dot when online prop is true', async () => {
        await act(() => tree.update(<ChatAvatar online={true} />));
        expect(tree.root.findByProps({ testID: 'online' })).toBeTruthy();
    });

    it('correctly applies width and height styles first case', async () => {
        await act(() => tree.update(<ChatAvatar width={16} height={16} />));
        expect([
            tree.root.findByType('Image').props.style.width,
            tree.root.findByType('Image').props.style.height
        ]).toEqual([16, 16]);
    });

    it('correctly applies width and height styles second case', async () => {
        await act(() =>
            tree.update(<ChatAvatar path="1234" width={16} height={16} />)
        );
        expect([
            tree.root.findByProps({ imageId: '1234' }).props.width,
            tree.root.findByProps({ imageId: '1234' }).props.height
        ]).toEqual([16, 16]);
    });
});
