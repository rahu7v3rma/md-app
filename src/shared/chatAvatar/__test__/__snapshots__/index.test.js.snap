// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test chatAvatar snapshot test 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/default_avatar.jpg",
      }
    }
    style={
      {
        "borderRadius": 0,
        "height": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;
